### DISCUSSION TITLE
How the "tunnel" is done between host and guess ?

### DISCUSSION THREAD

**Discussion by tr4l** [sakai135/wsl-vpnkit#159] [Q&A]
Hi,

I'm wondering how the tunneling between the host and the guest is done (on version 0.3x).

Looking at the doc, it seems that you use [gvisor-tap-vsock](https://github.com/containers/gvisor-tap-vsock) for that purpose, and that project use Qemu, Hyperkit, Hyper-V and User Mode Linux. BTW I discovered vsock which seems amazing.

Looking more in details, I've seen that you wrote your own [transport](https://github.com/sakai135/wsl-vpnkit/tree/main/pkg/transport)

Let me know if I'm correct now.
The transport you wrote use a feature of WSL which allow to run windows executable, on WSL console and "shared" the console.
The executable is run under windows context (so have access to windows network), and the output on the console can be seen on the wsl. With that you "just" wrap standard input and output of the console to communicate.

If this is correct, this is a gernius move.

**Comment by sakai135** [comment]
@tr4l You're right, I'm using the stdin/out integration in WSL to communicate between WSL and Windows. I got that idea from Docker for Windows, so credits go to that team.
